test/stark-proxy/sp3-owner.spec.ts

```diff
@@ -0,0 +1,145 @@
+          merkleDistributorAddress: merkleDistributor.address,
+          starkPerpetualAddress: starkPerpetual.address,
+          dydxCollateralTokenAddress: dydxToken.address,
+        })).starkProxyV1WithV3Impl;
+        const starkProxyV3: StarkProxyV3 = new StarkProxyV3__factory(deployer).attach(starkProxyV1WithV3Impl.address);
+
+        await Promise.all([
+          starkProxyV3.delegate(
+            dydxToken.address,
+            delegateeAddress,
+          ),
+          starkProxyV3.delegate(
+            safetyModule.address,
+            anotherDelegateeAddress,
+          ),
+        ]);
+
+        expect(
+          await dydxToken.getDelegateeByType(starkProxyV3.address, DelegationType.VOTING_POWER),
+        ).to.equal(delegateeAddress);
+        expect(
+          await dydxToken.getDelegateeByType(starkProxyV3.address, DelegationType.PROPOSITION_POWER),
+        ).to.equal(delegateeAddress);
+
+        expect(
+          await safetyModule.getDelegateeByType(starkProxyV3.address, DelegationType.VOTING_POWER),
+        ).to.equal(anotherDelegateeAddress);
+        expect(
+          await safetyModule.getDelegateeByType(starkProxyV3.address, DelegationType.PROPOSITION_POWER),
+        ).to.equal(anotherDelegateeAddress);
+      });
+
+      it('OWNER_ROLE can delegateByType DYDX and staked DYDX token', async () => {
+        const starkProxyV1WithV3Impl: StarkProxyV1 = (await deployStarkProxyV3({
+          liquidityStakingAddress: liquidityStaking.address,
+          merkleDistributorAddress: merkleDistributor.address,
+          starkPerpetualAddress: starkPerpetual.address,
+          dydxCollateralTokenAddress: dydxToken.address,
+        })).starkProxyV1WithV3Impl;
+        const starkProxyV3: StarkProxyV3 = new StarkProxyV3__factory(deployer).attach(starkProxyV1WithV3Impl.address);
+
+        await Promise.all([
+          starkProxyV3.delegateByType(
+            dydxToken.address,
+            delegateeAddress,
+            DelegationType.VOTING_POWER,
+          ),
+          starkProxyV3.delegateByType(
+            safetyModule.address,
+            anotherDelegateeAddress,
+            DelegationType.PROPOSITION_POWER,
+          ),
+        ]);
+
+        expect(
+          await dydxToken.getDelegateeByType(starkProxyV3.address, DelegationType.VOTING_POWER),
+        ).to.equal(delegateeAddress);
+        expect(
+          await dydxToken.getDelegateeByType(starkProxyV3.address, DelegationType.PROPOSITION_POWER),
+        ).to.equal(starkProxyV3.address);
+
+        expect(
+          await safetyModule.getDelegateeByType(starkProxyV3.address, DelegationType.VOTING_POWER),
+        ).to.equal(starkProxyV3.address);
+        expect(
+          await safetyModule.getDelegateeByType(starkProxyV3.address, DelegationType.PROPOSITION_POWER),
+        ).to.equal(anotherDelegateeAddress);
+      });
+
+      it('OWNER_ROLE can be set to another address', async () => {
```

Comment:
Can you add tests verifying the following:

• addresses without `OWNER_ROLE` cannot call the new functions
• the owner of the proxy admin contract is the short timelock
• all roles on the stark proxy contract are assigned correctly (and other addresses don't hold those roles). This is best done through events such as [`RoleGranted`](https://github.com/dydxfoundation/governance-contracts/blob/master/contracts/dependencies/open-zeppelin/AccessControl.sol#L87) and [`RoleRevoked`](https://github.com/dydxfoundation/governance-contracts/blob/master/contracts/dependencies/open-zeppelin/AccessControl.sol#L96).

After these are deployed and the governance proposal is passed, we won't be able to update these addresses without another governance proposal so we should ensure the contracts are configured correctly.